sdk/exporter/exporter.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+)
+
+// Type defines all the types of exporters we have available
+type Type int32
+
+// All the exporter types
+const (
+	ExporterTypeSync = iota
+	ExporterTypeBatch
+)
+
+// SyncExporter is a synchronous exporter
+type SyncExporter interface {
+	ExportSpan(context.Context, interface{})
+}
+
+// BatchExporter is a type for functions that receive sampled trace spans.
+//
+// The ExportSpans method is called asynchronously. However BatchExporter should
+// not take forever to process the spans.
+//
+// The SpanData should not be modified.
+type BatchExporter interface {
+	ExportSpans(context.Context, []interface{})
```

Comment:
In both ExportSpan(s) methods, I've currently set an interface, so the exporter can take multiple span structs (depending whether we're exporting to metrics or traces). This is causing a lot of weird things when having to turn SpanData back and forth to interfaces.
Maybe the SpanData struct should be moved to this package?

Reply:
Thinking this through, the interface definitely seems wrong, and a metrics exporter won't be sending SpanData anyway. So I believe the struct can be moved into this package.
I'll be doing that very soon.

Reply:
I agree on moving SpanData to this package and removing interface.